File: LeetCode/LeetCode Practice Problems/448- Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans;
        
        for(auto i: nums) {
            int v = abs(i) - 1;
            if(nums[v] > 0) {
                nums[v] = -nums[v];
            }
        }
        
        int pos = 1;
        for(auto i : nums) {
            if(i>0) {
                ans.push_back(pos);
            }
            
            pos++;
        }
        
        return ans;
        
        
    }
};

File: LeetCode/LeetCode Practice Problems/448-Find All Numbers Disappeared in an Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        List<Integer> ans = new ArrayList();
        for(int i=0;i<n;i++) {
            int val = Math.abs(nums[i]) - 1;
            if(nums[val] > 0) {
                nums[val] *= -1;
            }
        }
        for(int i=0;i<n;i++) {
            if(nums[i] > 0) {
                ans.add(i + 1);
            }
        }
        return ans;
    }
}
